packages/runtime/container-runtime/src/opLifecycle/remoteMessageProcessor.ts

```diff
@@ -49,65 +49,49 @@ export class RemoteMessageProcessor {
 	 * a singleton array [remoteMessageCopy] is returned
 	 */
 	public process(
-		remoteMessageCopy: ISequencedDocumentMessage,
+		remoteMessage: ISequencedDocumentMessage,
 	): InboundSequencedContainerRuntimeMessageOrSystemMessage[] {
-		const result: InboundSequencedContainerRuntimeMessageOrSystemMessage[] = [];
-
-		ensureContentsDeserialized(remoteMessageCopy);
-
-		// Ungroup before and after decompression for back-compat (cleanup tracked by AB#4371)
-		for (const ungroupedMessage of this.opGroupingManager.ungroupOp(remoteMessageCopy)) {
-			const message = this.opDecompressor.processMessage(ungroupedMessage).message;
-
-			for (let ungroupedMessage2 of this.opGroupingManager.ungroupOp(message)) {
-				// unpack and unchunk the ungrouped message in place
-				unpackRuntimeMessage(ungroupedMessage2);
-				const chunkProcessingResult =
-					this.opSplitter.processRemoteMessage(ungroupedMessage2);
-				ungroupedMessage2 = chunkProcessingResult.message;
-
-				// if the splitter is still rebuilding the original message, there is no need to continue processing
-				if (chunkProcessingResult.state === "Accepted") {
-					continue;
-				}
-
-				// If the message is not chunked there is no need to continue processing
-				if (chunkProcessingResult.state !== "Processed") {
-					result.push(
-						ungroupedMessage2 as InboundSequencedContainerRuntimeMessageOrSystemMessage,
-					);
-					continue;
-				}
-
-				// Ungroup before and after decompression for back-compat (cleanup tracked by AB#4371)
-				for (const ungroupedMessageAfterChunking of this.opGroupingManager.ungroupOp(
-					ungroupedMessage2,
-				)) {
-					const decompressionAfterChunking = this.opDecompressor.processMessage(
-						ungroupedMessageAfterChunking,
-					);
-
-					for (const ungroupedMessageAfterChunking2 of this.opGroupingManager.ungroupOp(
-						decompressionAfterChunking.message,
-					)) {
-						if (decompressionAfterChunking.state === "Skipped") {
-							// After chunking, if the original message was not compressed,
-							// there is no need to continue processing
-							result.push(
-								ungroupedMessageAfterChunking2 as InboundSequencedContainerRuntimeMessageOrSystemMessage,
-							);
-							continue;
-						}
-
-						// The message needs to be unpacked after chunking + decompression
-						unpack(ungroupedMessageAfterChunking2);
-						result.push(ungroupedMessageAfterChunking2);
-					}
-				}
+		let message = remoteMessage;
+		ensureContentsDeserialized(message);
+
+		// Checking for compression needs to happen before unpacking the message
```

Comment:
Can you add a big fat comment at the top here explaining the various states the incoming op could be in, and how this will deal with all those perumtations?

Also, it's hard for me to recognize how this covers all 6 permutations.  Before I could trace through it with the nesting, but I'm not seeing the symmetry I expected.  Hopefully that comment would help someone like me understand.